### PR TITLE
Giving life to Orpheus ! - Added chatbot functionality.

### PR DESCRIPTION
**Pull request by OptimalBlock489**
I set out on a mission to give life to Orpheus 5 days ago. I created a fine-tuned model with Orpheus's personality. This pr has some additional code that makes Orpheus send an api request to OPENAI whenever it finds the word Orpheus or Orpheus-GPT in a text.

### CODE REVIEW

**Comment by OptimalBlock489** [comment]
Somebody please add OPENAI_API_KEY as an environment variable for it to not break.

File: src/interactions/gpt.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    `User: Actually, I'm a bit stuck on a coding problem. Do you think you could help me out?`,
+    `Orpheus: Of course! I'm always happy to help out my fellow hackers. What's the problem you're facing? Maybe I can help you figure it out.`,
+    `User: I'm having trouble with this piece of code. It keeps giving me errors and I can't figure out why.`,
+    `Orpheus: Hmm, that's definitely a tricky problem. Have you tried checking your syntax and making sure all your variables are properly defined? Sometimes that can cause errors. If that doesn't work, we can dive deeper into the problem and see what's really going on.`,
+    `User: Thanks, Orpheus! You're the best. Do you have any tips for learning coding faster?`,
+    `Orpheus: Well, one thing that's helped me is practicing coding challenges and pushing myself to try new things. It can also be really helpful to work with other people and bounce ideas off of each other. And of course, never give up! Coding can be tough sometimes, but it's always worth it in the end."`,
+    `Now the real user sends a message, with reference to the prompt and your knowledge, respond in the best way possible.\n
```
Comment (sampoder) [inline src/interactions/gpt.js:37]: What do the responses look like using this prompt?

Comment (OptimalBlock489) [reply src/interactions/gpt.js:37]: This is a real conversation I had right now with this exact prompt design. :

User: Hi there, orpheus !
Orpheus: Hey there! I've been up to all sorts of things lately. Just the other day, I was learning about new coding languages and trying to figure out how to apply them in my adventures. What about you? Have you been working on any coding projects recently?
User: I have been quite busy with school, although I have an idea for a project which I will start working on.
Orpheus: That's great! I'm always happy to help out my fellow hackers. What's the problem you're facing? Maybe I can help you figure it out.
User: What kind of resources do you suggest for learning next.js ?
Orpheus: The best resource for learning next.js is the official website, which contains a comprehensive guide to getting started. You can also check out the official docs for more information. And if you get stuck, you can always ask for help in the official Slack channel.
Orpheus: Another great resource is the official Slack channel, where you can connect with fellow hackers and get feedback on your projects. And if you're looking for a coding challenge, you can check out Hack Club's coding challenges page, which contains a list of over 200 coding projects that you can build and improve over time.

**Comment by OptimalBlock489** [comment]
@max

File: src/interactions/gpt.js (hunk 1 above)
Comment (cytronicoder) [reply src/interactions/gpt.js:37]: I don't think this is accurate - there's no official Slack channel for Next.js
Can we maybe work on this?

**Review by cytronicoder — CHANGES_REQUESTED** [review]
Maybe the model should be more informal and chat-based (i.e., encourages them to "chat" with the bot for a better final response)? I don't know, but this sounds a lot like ChatGPT, and I don't think we would want that.

**Comment by OptimalBlock489** [comment]
I actually plan refining the prompt after deploying it, see how it works. Testing this on playground is really inefficient.

**Comment by OptimalBlock489** [comment]
The model will always hallucinate about topics it doesn't know about. It's a nature of GPT3. GPT 3.5 would just say I don't know about that but davinci will make up things. We could reduce the temperature to .15 or .1 but it's already at .2, anymore and it could hamper that "chat" behaviour we are looking for.

**Comment by maxwofford** [comment]
@OptimalBlock489 Can we provide example discussion that would help with its chatter? I can def provide a list.

**Comment by OptimalBlock489** [comment]
> @OptimalBlock489 Can we provide example discussion that would help with its chatter? I can def provide a list.

I have already provided an example discussion in the prompt of gpt.js

**Comment by OptimalBlock489** [comment]
We can edit the prompt. But keep in mind that this generation of GPT models only accept 2048 tokens. My current prompt in 850ish tokens, so the result can only go upto 1500tokens (850[context] +1500 [output] = 2350 tokens, [the output is not always 1500 tokens so it's safe]). Enlarging the the prompt will result in smaller output constraints and longer processing time. 😄

**Comment by maxwofford** [comment]
@kognise do you have access to the openapi account? Having trouble getting in– could you generate a token and get it in production for this?

**Comment by maxwofford** [comment]
@OptimalBlock489 know this has been dragging on for a while now– how about we conditionally check if the env is set and if not skip this interaction trigger. That way once it's set the commands will be usable.

**Comment by kognise** [comment]
Environment variable added and redeployed master, is there anything else we need to do to get this working?

**Comment by OptimalBlock489** [comment]
> Environment variable added and redeployed master, is there anything else we need to do to get this working?

Doesn't seem to be working, can we peek at logs ?

**Comment by OptimalBlock489** [comment]
> > Environment variable added and redeployed master, is there anything else we need to do to get this working?
> 
> Doesn't seem to be working, can we peek at logs ? 

@kognise does the api key belong to hackclub's organization account ?

**Comment by kognise** [comment]
Yes, it does belong to the org account. Not seeing anything in logs.

**Comment by maxwofford** [comment]
```
2023-05-02T20:18:15.259876+00:00 app[web.1]: Middleware: I'm scrying a indirect_mention in my crystal ball
2023-05-02T20:18:15.262809+00:00 app[web.1]: An error occurred in the receive middleware:  TypeError: Assignment to constant variable.
2023-05-02T20:18:15.262810+00:00 app[web.1]: at interactionGPT (/app/build/interactions/gpt.js:38:8)
2023-05-02T20:18:15.262810+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:1348:32
2023-05-02T20:18:15.262810+00:00 app[web.1]: at next (/app/node_modules/ware/lib/index.js:82:27)
2023-05-02T20:18:15.262810+00:00 app[web.1]: at Ware.run (/app/node_modules/ware/lib/index.js:88:3)
2023-05-02T20:18:15.262811+00:00 app[web.1]: at Object.<anonymous> (/app/node_modules/botkit/lib/CoreBot.js:1347:49)
2023-05-02T20:18:15.262811+00:00 app[web.1]: at Botkit.botkit.trigger (/app/node_modules/botkit/lib/CoreBot.js:1396:47)
2023-05-02T20:18:15.262812+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:148:32
2023-05-02T20:18:15.262812+00:00 app[web.1]: at bot.findConversation (/app/node_modules/botkit/lib/Slackbot_worker.js:923:9)
2023-05-02T20:18:15.262812+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:144:21
2023-05-02T20:18:15.262812+00:00 app[web.1]: at next (/app/node_modules/ware/lib/index.js:82:27)
```

**Comment by maxwofford** [comment]
Caused by [this message](https://hackclub.slack.com/archives/C0P5NE354/p1683058693500629)

**Comment by OptimalBlock489** [comment]
> ```
> 2023-05-02T20:18:15.259876+00:00 app[web.1]: Middleware: I'm scrying a indirect_mention in my crystal ball
> 2023-05-02T20:18:15.262809+00:00 app[web.1]: An error occurred in the receive middleware:  TypeError: Assignment to constant variable.
> 2023-05-02T20:18:15.262810+00:00 app[web.1]: at interactionGPT (/app/build/interactions/gpt.js:38:8)
> 2023-05-02T20:18:15.262810+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:1348:32
> 2023-05-02T20:18:15.262810+00:00 app[web.1]: at next (/app/node_modules/ware/lib/index.js:82:27)
> 2023-05-02T20:18:15.262810+00:00 app[web.1]: at Ware.run (/app/node_modules/ware/lib/index.js:88:3)
> 2023-05-02T20:18:15.262811+00:00 app[web.1]: at Object.<anonymous> (/app/node_modules/botkit/lib/CoreBot.js:1347:49)
> 2023-05-02T20:18:15.262811+00:00 app[web.1]: at Botkit.botkit.trigger (/app/node_modules/botkit/lib/CoreBot.js:1396:47)
> 2023-05-02T20:18:15.262812+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:148:32
> 2023-05-02T20:18:15.262812+00:00 app[web.1]: at bot.findConversation (/app/node_modules/botkit/lib/Slackbot_worker.js:923:9)
> 2023-05-02T20:18:15.262812+00:00 app[web.1]: at /app/node_modules/botkit/lib/CoreBot.js:144:21
> 2023-05-02T20:18:15.262812+00:00 app[web.1]: at next (/app/node_modules/ware/lib/index.js:82:27)
> ```

Could the code be buggy ? I can't find any bugs.

**Pull request merged**